flow/Arena.h

```diff
@@ -208,26 +209,32 @@ inline void save( Archive& ar, const Arena& p ) {
 	// No action required
 }
 
+// Optional is a wrapper for std::optional. There
+// are two primary reasons to use this wrapper instead
+// of using std::optional directly:
+//
+// 1) Legacy: A lot of code was written using Optional before
+//    std::optional was available.
+// 2) When you call get but no value is present Optional gives an
+//    assertion failure. std::optional, on the other hand, would
+//    throw std::bad_optional_access.
 template <class T>
 class Optional : public ComposedIdentifier<T, 0x10> {
 public:
-	Optional() : valid(false) { memset(&value, 0, sizeof(value)); }
-	Optional(const Optional<T>& o) : valid(o.valid) {
-		if (valid) new (&value) T(o.get());
-	}
+	Optional() = default;
 
 	template <class U>
-	Optional(const U& t) : valid(true) { new (&value) T(t); }
+	Optional(const U& t) : impl(t) {}
```

Comment:
After comparing this to the `std::optional` constructors, I'm not sure what this does.

Ignoring the old version of this method, it looks like it's going to call:
```
template < class U = value_type >constexpr optional( U&& value );
```
since all the other single argument `std::optional` constructors want `U` to  be an `optional<....>` here.  To preserve the old behavior, do you need to add an `in_place_t`?